camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/EtlCounts.java

```diff
@@ -238,21 +248,40 @@ private void produceCount(String brokerList, ArrayList<byte[]> monitorSet) {
     props.put("request.required.acks", "1");
     props.put("request.timeout.ms", "30000");
     log.debug("Broker list: " + brokerList);
-    Producer producer = new Producer(new ProducerConfig(props));
+
+    Producer producer = createProducer(props);
+
     try {
       for (byte[] message : monitorSet) {
-        KeyedMessage keyedMessage = new KeyedMessage("TrackingMonitoringEvent", message);
-        producer.send(keyedMessage);
+        for (int i = 0; i < NUM_TRIES_PUBLISH_COUNTS; i++) {
+          try {
+            KeyedMessage keyedMessage = new KeyedMessage("TrackingMonitoringEvent", message);
+            producer.send(keyedMessage);
+            break;
+          } catch (Exception e) {
+            log.error("Publishing count for topic " + topic + " to " + brokerList.toString() + " has failed " + (i + 1)
+                + " times. " + (NUM_TRIES_PUBLISH_COUNTS - i - 1) + " more attempts will be made.");
+            if (i == NUM_TRIES_PUBLISH_COUNTS - 1) {
+              throw new RuntimeException(e.getMessage() + ": " + "Have retried maximum ("
+                  + NUM_TRIES_PUBLISH_COUNTS + ") times.");
+            }
+            try {
+              Thread.sleep((long)(Math.random() * (i + 1) * 1000));
+            } catch (InterruptedException e1) {
+              log.error("Caught interrupted exception between retries of publishing counts to Kafka. "
+                  + e1.getMessage());
+            }
+          }
+        }
       }
-    } catch (Exception e) {
-      e.printStackTrace();
-      System.out.println(topic + " issue sending tracking to " + brokerList.toString());
     } finally {
       if (producer != null) {
         producer.close();
       }
     }
-
   }
 
+  protected Producer createProducer(Properties props) {
+    return new Producer(new ProducerConfig(props));
+  }
```

Comment:
put `new Producer(new ProducerConfig(props))` as a method, so that it can be overridden in subclass EtlCountsForUnitTest.